02_classes-objects-methods/02_02_planets.py

```diff
@@ -2,5 +2,22 @@
 # a planet object.
 # Use the appropriate dunder method to get informative output with `print()`
 
-class Planet():
-    pass
+class Planet:
+    def __init__(self, planet, color) -> None:
+        self.planet = planet
+        self.color = color
+
+    def planetcolor(self):
+        return f'Plaent {self.planet} is {self.color}'
+
+    def __str__(self) -> str:
+        return f'{self.planet} - {self.color}'
+    
+    def __repr__(self) -> str:
+        return f'planet = {self.planet} \ncolor = {self.color}'
+
+e = Planet('earth' , 'blue')
+print(e.planetcolor())
+print(str(e))
+print(repr(e))
+
```

Comment:
so here you would not need to call `rep()` or `str()` manually. You can just do `print(e)` and it would work.

Reply:
well I guess if you have both a `__str__` and a `__repr__` method they compete a bit (print can only call one). Print seems to work on the `__str__` case here. And the repr needs to be called if you want to see it in action.